### PR TITLE
ftplugin(asm): add Matchit support

### PR DESCRIPTION
**Pull request by Freed-Wu**
Refer https://github.com/vim/vim/blob/master/runtime/ftplugin/masm.vim#L18-L29

### CODE REVIEW

File: runtime/ftplugin/asm.vim [hunk 1]
```diff
@@ -3,11 +3,21 @@
 " Maintainer:	Colin Caine <cmcaine at the common googlemail domain>
 " Last Change:  23 May 2020
 " 		2023 Aug 28 by Vim Project (undo_ftplugin)
+" 		2024-04-09
```
Comment (chrisbra) [inline runtime/ftplugin/asm.vim]: by Vim Project (add matchit support)

Comment (Freed-Wu) [reply runtime/ftplugin/asm.vim]: Fixed

File: runtime/ftplugin/asm.vim [hunk 2]
```diff
@@ -3,11 +3,21 @@
 " Maintainer:	Colin Caine <cmcaine at the common googlemail domain>
 " Last Change:  23 May 2020
 " 		2023 Aug 28 by Vim Project (undo_ftplugin)
+" 		2024-04-09 by Vim Project (add Matchit support)
```
Comment (dkearns) [inline runtime/ftplugin/asm.vim]: And then there were three date formats.

Can we please use YYYY Mon DD for the "Vim Project" lines?

File: runtime/ftplugin/asm.vim [hunk 3]
```diff
@@ -3,11 +3,21 @@
...
 
 if exists("b:did_ftplugin") | finish | endif
 let b:did_ftplugin = 1
 
+setl include=^\s*%\s*include
```
Comment (dkearns) [inline runtime/ftplugin/asm.vim]:
```suggestion
setl include=^\\s*%\\s*include
```

**Review by dkearns — CHANGES_REQUESTED** [review]

**Pull request closed without merging**